assets/node-resolver/update-node-resolver.sh

```diff
@@ -38,7 +38,12 @@ while true; do
   # Stale entries could exist in /etc/hosts if the service is deleted
   if [[ -n "${svc_ips[*]-}" ]]; then
     # Build a new hosts file from /etc/hosts with our custom entries filtered out
-    grep -v "# ${OPENSHIFT_MARKER}" "${HOSTS_FILE}" > "${TEMP_FILE}"
+    sed --unbuffered --silent "/# ${OPENSHIFT_MARKER}/d; w ${TEMP_FILE}" "${HOSTS_FILE}"
+    if [[ "$?" -ne 0 ]]; then
```

Comment:
That's probably not needed. I thought I saw a few occurrences where the pattern was printed (prior to silencing sed) but not written to file. With buffering disabled I couldn't reproduce that behavior but now I can't reproduce that behavior with buffering enabled so it may have been a wrong interpretation.